### PR TITLE
Feat: support url in ref-objects

### PR DESCRIPTION
**Pull request by Somefive**
Signed-off-by: Somefive <yd219913@alibaba-inc.com>

### Description of your changes

Support referring objects in remote file by using ref-objects.

#### Example
```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: example-app
  namespace: default
spec:
  components:
  - name: busybox
    type: ref-objects
    properties:
      urls: ["https://gist.githubusercontent.com/Somefive/b189219a9222eaa70b8908cf4379402b/raw/e603987b3e0989e01e50f69ebb1e8bb436461326/example-busybox-deployment.yaml"]
```

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]

File: pkg/appfile/parser.go [hunk 1]
```diff
@@ -342,13 +343,26 @@ func (p *Parser) parseReferredObjects(ctx context.Context, af *Appfile) error {
 		if err := utils.StrictUnmarshal(comp.Properties.Raw, spec); err != nil {
 			return errors.Wrapf(err, "invalid properties for ref-objects in component %s", comp.Name)
 		}
+		if len(spec.Objects) > 0 && len(spec.URLs) > 0 {
+			return fmt.Errorf("cannot use objects and urls in ref-objects at the same time")
+		}
```
Comment (wonderflow) [inline pkg/appfile/parser.go:348]: why not, just append them together into the slice.

**Review by wonderflow — COMMENTED** [review]
where's the step generator check for deploy step?

File: pkg/appfile/parser.go (hunk 1 above)
Comment (Somefive) [reply pkg/appfile/parser.go]: Removed.

**Comment by Somefive** [comment]
> where's the step generator check for deploy step?

https://github.com/kubevela/kubevela/blob/92b6860737656d6fca5d0f4fc664e5d21b73154f/pkg/workflow/step/generator.go#L159

**Review by FogDong — APPROVED** [review]

**Pull request merged**